pkg/apis/cluster/zz_generated.api.register.go

```diff
@@ -188,85 +196,77 @@ type MachineSpec struct {
-}
-
 type MachineVersionInfo struct {
 	Kubelet      string
 	ControlPlane string
 }
 
-type ClusterNetworkingConfig struct {
-	Services      NetworkRanges
-	Pods          NetworkRanges
-	ServiceDomain string
+type ProviderConfig struct {
+	Value     *pkgruntime.RawExtension
+	ValueFrom *ProviderConfigSource
 }
 
 // +genclient
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-type MachineDeployment struct {
+type Machine struct {
 	metav1.TypeMeta
 	metav1.ObjectMeta
-	Spec   MachineDeploymentSpec
-	Status MachineDeploymentStatus
-}
-
-type NetworkRanges struct {
-	CIDRBlocks []string
-}
-
-type MachineDeploymentStatus struct {
-	ObservedGeneration  int64
-	Replicas            int32
-	UpdatedReplicas     int32
-	ReadyReplicas       int32
-	AvailableReplicas   int32
-	UnavailableReplicas int32
-}
-
-type MachineDeploymentSpec struct {
-	Replicas                *int32
-	Selector                metav1.LabelSelector
-	Template                MachineTemplateSpec
-	Strategy                MachineDeploymentStrategy
-	MinReadySeconds         *int32
-	RevisionHistoryLimit    *int32
-	Paused                  bool
-	ProgressDeadlineSeconds *int32
+	Spec   MachineSpec
+	Status MachineStatus
 }
 
-type MachineDeploymentStrategy struct {
-	Type          clustercommon.MachineDeploymentStrategyType
-	RollingUpdate *MachineRollingUpdateDeployment
+type ProviderConfigSource struct {
 }
 
-type MachineRollingUpdateDeployment struct {
-	MaxUnavailable *utilintstr.IntOrString
-	MaxSurge       *utilintstr.IntOrString
+type MachineStatus struct {
+	NodeRef        *corev1.ObjectReference
+	LastUpdated    metav1.Time
+	Versions       *MachineVersionInfo
+	ErrorReason    *clustercommon.MachineStatusError
+	ErrorMessage   *string
+	ProviderStatus *pkgruntime.RawExtension
+	Addresses      []corev1.NodeAddress
+	Conditions     []corev1.NodeCondition
 }
 
 // +genclient
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-type Machine struct {
+type Cluster struct {
 	metav1.TypeMeta
 	metav1.ObjectMeta
-	Spec   MachineSpec
-	Status MachineStatus
+	Spec   ClusterSpec
+	Status ClusterStatus
 }
 
-type MachineStatus struct {
-	NodeRef        *corev1.ObjectReference
-	LastUpdated    metav1.Time
-	Versions       *MachineVersionInfo
-	ErrorReason    *clustercommon.MachineStatusError
-	ErrorMessage   *string
+type ClusterSpec struct {
+	ClusterNetwork ClusterNetworkingConfig
+	ProviderConfig ProviderConfig
+}
+
+type ClusterStatus struct {
+	APIEndpoints   []APIEndpoint
+	ErrorReason    clustercommon.ClusterStatusError
+	ErrorMessage   string
 	ProviderStatus *pkgruntime.RawExtension
-	Addresses      []corev1.NodeAddress
-	Conditions     []corev1.NodeCondition
+}
+
+type APIEndpoint struct {
+	Host string
+	Port int
+}
+
+type ClusterNetworkingConfig struct {
+	Services      NetworkRanges
+	Pods          NetworkRanges
+	ServiceDomain string
+}
+
+type NetworkRanges struct {
+	CIDRBlocks []string
```

Comment:
I'm having difficulty telling if there is an actual diff here outside of the generation script re-ordering the structs. If there is no actual diff, can you replace this with the existing upstream version?

Reply:
Since you didn't change api types, there shouldn't be any relevant changes in this file. Please remove from the PR.